src/codegen/enums.rs

```diff
@@ -0,0 +1,158 @@
+use analysis::namespaces;
+use case::CaseExt;
+use codegen::general::{self, version_condition, version_condition_string};
+use config::gobjects::GObject;
+use env::Env;
+use file_saver;
+use library::*;
+use std::collections::HashSet;
+use std::io::prelude::*;
+use std::io::Result;
+use std::path::Path;
+use traits::*;
+use version::Version;
+
+pub fn generate(env: &Env, root_path: &Path, mod_rs: &mut Vec<String>) {
+    let path = root_path.join("enums.rs");
+    file_saver::save_to_file(path, env.config.make_backup, |w| {
+        try!(general::start_comments(w, &env.config));
+        try!(writeln!(w, "{}", "
+use ffi;
+use glib_ffi;
+use glib::error::ErrorDomain;
+use glib::translate::*;
+"));
+
+        let configs = env.config.objects.values()
+            .filter(|c| { 
+                c.status.need_generate() &&
+                    c.type_id.map_or(false, |tid| tid.ns_id == namespaces::MAIN)
+            });
+        let mut first = true;
+        for config in configs {
+            if let Type::Enumeration(ref enum_) = *env.library.type_(config.type_id.unwrap()) {
+                if first {
+                    mod_rs.push("\nmod enums;".into());
+                    first = false;
+                }
+                if let Some (cfg) = version_condition_string(env, enum_.version, false, 0) {
```

Comment:
I thinking about `version_condition_string_default(env, enum_.version)` here, as too many `(... ,false, 0)` in project.